types/cuint/index.d.ts

```diff
@@ -0,0 +1,34 @@
+// Type definitions for cuint 0.2
+// Project: https://github.com/pierrec/js-cuint
+// Definitions by:  Lukas Tetzlaff <https://github.com/ltetzlaff>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export type UIntFactory<T extends UInt> = (
+    ...numbers: Array<number | string>
+) => T;
+
+export const UINT64: UIntFactory<UInt64>;
+export const UINT32: UIntFactory<UInt32>;
+
+export class UInt {
```

Comment:
`UInt` is declared here as both a type and a value because it is defined as a `class`, however the value-side isn't actually an export you can reference.

For these definitions, you might be better with something like:

```ts
export interface Uint {
    // ... every member currently in `UInt` except `constructor` ...
}

export interface UintConstructor {
  // Uint called as a function:
  (value: number): Uint;
  (low: number, high: number): Uint;
  (text: string, radix?: number): Uint;

  // Uint called as a constructor:
  new (value: number): Uint;
  new (low: number, high: number): Uint;
  new (text: string, radix?: number): Uint;

  prototype: Uint;
}

export const UINT64: UintConstructor;
export const UINT32: UintConstructor;
```

Reply:
Cool, thx for the comment 👍

I maintained your suggestions from above while keeping 64-bit and 32-bit separate through generics to not make them interchangeable when using the lib.